packages/@react-aria/focus/src/FocusScope.tsx

```diff
@@ -608,7 +608,7 @@ function useRestoreFocus(scopeRef: RefObject<Element[] | null>, restoreFocus?: b
       }
 
       let focusedElement = ownerDocument.activeElement as FocusableElement;
-      if (!isElementInScope(focusedElement, scopeRef.current)) {
+      if (!isElementInChildScope(focusedElement, scopeRef) || !shouldRestoreFocus(scopeRef)) {
```

Comment:
FocusScope changes are so that focus restoration works across React portals. The difference here is that `isElementInChildScope` checks whether the element is either a descendant of this focus scope in the DOM, or any other child scope (which might be in a portal).

Reply:
do any of our tests cover the changes in this file indirectly?